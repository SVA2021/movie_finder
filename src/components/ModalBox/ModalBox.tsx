import CloseIcon from '@mui/icons-material/Close';
import {Box, Button} from '@mui/material';
import {FC, memo} from 'react';

interface ModalBoxProps {
  closeHandler: () => void
  children: any
}

export const ModalBox: FC<ModalBoxProps> = memo(({children, closeHandler}) => {
  return (
    <Box
      data-testid={'modal-box'}
      margin={'0 auto'}
      p={{xs: 1, md: 2, lg: 4}}
      zIndex={1000}
      bgcolor={'primary.main'}
      sx={{
        position: 'fixed',
        inset: 0,
        height: '100vh',
        width: '100vw',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
      }}
    >
      {children}
      <Box
        sx={{
          position: 'absolute',
          top: {xs: 10, md: 32},
          right: {xs: 10, md: 32}
        }}>
        <Button onClick={closeHandler} >
          <CloseIcon fontSize='small' />
        </Button>
      </Box>
    </Box>
  );
});