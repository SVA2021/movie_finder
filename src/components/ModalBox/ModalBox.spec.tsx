import {render, screen, fireEvent} from "@testing-library/react";
import {ModalBox} from "./ModalBox";

describe('render ModalBox', () => {

  const onClick = jest.fn();
  const setUp = () => render(<ModalBox closeHandler={onClick} >children</ModalBox>);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('modal-box')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/children/i)).toBeInTheDocument();
  });

  it('check button', () => {
    setUp();
    fireEvent.click(screen.getByRole('button'))
    expect(onClick).toBeCalled();
  });

})