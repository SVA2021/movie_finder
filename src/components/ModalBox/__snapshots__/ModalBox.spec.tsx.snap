// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ModalBox snapshot test 1`] = `
<div
  class="MuiBox-root css-n3arjh"
  data-testid="modal-box"
>
  children
  <div
    class="MuiBox-root css-ri25m1"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
